technologydata/technologies.py

```diff
@@ -288,63 +292,46 @@ def adjust_scale(
 
         """
         # Default parameters
-        parameters = (
-            parameters
-            if parameters
-            else [
-                "investment",
-                "capex",
-                "opex",
-            ]
-        )
-
-        changed_data = None
-        unchanged_data = None
+        parameters = parameters or ["investment", "capex", "opex"]
 
+        # Filter data based on parameters
         changed_data = self.data.query(f"parameter in {parameters}")
         unchanged_data = self.data.loc[~self.data.index.isin(changed_data.index)]
 
+        # Check for missing scale values
         if changed_data["scale"].isna().any():
             logger.warning(
                 "Some data entries have no associated `scale` and will yield NaN values after scaling. "
                 "Set their `scale` or remove them from the collection before using this function."
             )
 
         # TODO create a copy with converted units that align with the requested scale
-        # and use this data for calculating the scaling factors, but don't use the changed values for returning, rather the original units
+        # and use this data for calculating the scaling factors,
+        # but don't use the changed values for returning, rather the original units
+
+        # Check for unit compatibility
         if any(changed_data["scale_unit"] != unit):
             raise NotImplementedError(
                 f"Unit conversion for different units to {unit} is not implemented yet."
             )
 
-        # Calculate the scaling factor, where we assume that all entries are specific values
-        changed_data["scaling_factor"] = (new_scale / changed_data["scale"]) ** (
-            scaling_exponent - 1
-        )
-
-        # Apply the scaling factor
-        changed_data["value"] = changed_data["value"] * changed_data["scaling_factor"]
-
-        # Remove obsolete scaling_factor column
-        changed_data = changed_data.drop(columns=["scaling_factor"])
+        # Calculate the scaling factor
+        scaling_factor = (new_scale / changed_data["scale"]) ** (scaling_exponent - 1)
 
-        # Add information about new scale + unit
-        changed_data["scale"] = new_scale
-        changed_data["scale_unit"] = unit
+        # Create a new DataFrame for adjusted values
+        adjusted_data = changed_data.copy()
```

Comment:
Why create a copy of `changed_data` when it is not used afterwards anymore? @finozzifa